tests/potential/test_potential.py

```diff
@@ -308,5 +309,281 @@ def test_fromfile(self):
+        f = p.force(pair=(('1','1')), r=[0.25,1.75])
+        np.testing.assert_allclose(f, [-2.0,0.0])
+
+        #test with rmin and rmax set
+        p = test_PairPotential.LinPot(types=('1',), params=('m','rmin','rmax'),
+                                      default={'m':2,'rmin':0.0,'rmax':1.0})
+        f = p.force(pair=(('1','1')), r=0.5)
+        self.assertEqual(f, -2.0)
+        f = p.force(pair=(('1','1')), r=[-0.5,0.0,1.0,1.5])
+        np.testing.assert_allclose(f, [0.0,-2.0,-2.0,0.0])
+
+        #test with shift set
+        p = test_PairPotential.LinPot(types=('1',), params=('m','rmax','shift'),
+                                      default={'m':2,'rmax':1.0,'shift':True})
+        f = p.force(pair=(('1','1')), r=0.5)
+        self.assertEqual(f, -2.0)
+        f = p.force(pair=(('1','1')), r=[1.0,1.5])
+        np.testing.assert_allclose(f, [-2.0,0.0])
+
+    def test_derivative(self):
+        """Test derivative method"""
+        #test with no cutoffs
+        p = test_PairPotential.LinPot(types=('1',), params=('m',),
+                                      default={'m':2})
+        d = p.derivative(pair=(('1','1')), param='m', r=0.5)
+        self.assertEqual(d, 0.5)
+        d = p.derivative(pair=(('1','1')), r=[0.25,0.75], param='m')
+        np.testing.assert_allclose(d, [0.25,0.75])
+
+        #test with rmin set
+        p = test_PairPotential.LinPot(types=('1',), params=('m','rmin'),
+                                      default={'m':2,'rmin':0.0})
+        d = p.derivative(pair=(('1','1')), param='m', r=0.0)
+        self.assertEqual(d, 0.0)
+        d = p.derivative(pair=(('1','1')), param='m', r=[-0.25,0.75])
+        np.testing.assert_allclose(d, [0.0,0.75])
+
+        #test with rmax set
+        p = test_PairPotential.LinPot(types=('1',), params=('m','rmax'),
+                                      default={'m':2,'rmax':1.0})
+        d = p.derivative(pair=(('1','1')), param='m', r=1.0)
+        self.assertEqual(d, 1.0)
+        d = p.derivative(pair=(('1','1')), param='m', r=[0.25,1.75])
+        np.testing.assert_allclose(d, [0.25,0.0])
+
+        #test with rmin and rmax set
+        p = test_PairPotential.LinPot(types=('1',), params=('m','rmin','rmax'),
+                                      default={'m':2,'rmin':0.0,'rmax':1.0})
+        d = p.derivative(pair=(('1','1')), param='m', r=0.5)
+        self.assertEqual(d, 0.5)
+        d = p.derivative(pair=(('1','1')), param='m', r=[-0.5,0.0,1.0,1.5])
+        np.testing.assert_allclose(d, [0.0,0.0,1.0,0.0])
+
+        #test with shift set
+        p = test_PairPotential.LinPot(types=('1',), params=('m','rmax','shift'),
+                                      default={'m':2,'rmax':1.0,'shift':True})
+        d = p.derivative(pair=(('1','1')), param='m', r=0.5)
+        self.assertEqual(d, 0.5)
+        d = p.derivative(pair=(('1','1')), param='m', r=[1.0,1.5])
+        np.testing.assert_allclose(d, [1.0,0.0])
+
+    def test_iteration(self):
```

Comment:
What is the goal of this test? It seems like it's duplicating the CoefficientMatrix tests since everything is testing on `p.coeff` rather than `p` itself?